### DISCUSSION TITLE
Trouble getting it to run

### DISCUSSION THREAD

**Discussion by mjank72** [jertel/vuegraf#87] [Q&A] [answered]
I'm new to this and am not a Python developer.  That said, I'm hoping the issue I'm running into is a simple fix.  I'm running on Windows 10 and have installed Python 3.10.2 and InfluxDB2 v2.2.1.  I created a config file (vuegraf.json) based on the example and modified for my config (below)

{
    "influxdb": {
        "version": 2,
        "url": "http://localhost:8086",
        "org": "InfluxOrg",
        "bucket": "EmporiaData",
        "token": "InfluxUserAPIToken",
        "reset": false
    },
    "accounts": [
        {
            "name": "Primary Residence",
            "email": "my@email",
            "password": "passwd",
            "devices": [
                {
                    "name": "Home Main and Circuits",
                    "channels": [
                        "Outside A/C (Attic)",
                        "Outside A/C (Basement)",
                        "Basement Furnace",
                        "Sun Room",
                        "Kitchen 1",
                        "Kitchen 2",
                        "Family Room",
                        "Master Bedroom",
			"Bedroom (Adrian)",
			"Bedroom (Alex)",
			"Bedroom (Kaleigh)",
			"Bedroom Basement",
			"Rec Room Plugs",
			"Rec Room",
			"Washer",
			"Attic Furnace"
                    ]
                }
            ]
        }
    ]
}
****************************************************************************************************************************************
EDIT

I figured out the first issue below (forgot to install the dependencies), but now I am seeing a different issue.

     Traceback (most recent call last):
       File "c:\Program Files\InfluxData\vuegraf-master\src\vuegraf\vuegraf.py", line 207, in <module>
         signal.signal(signal.SIGHUP, handleExit)
     AttributeError: module 'signal' has no attribute 'SIGHUP'

Any thoughts?  Thanks!!

**************************************************************************************************************************************
When I try to execute the Python script I get the following error message:
     Traceback (most recent call last):
         File "c:\Program Files\InfluxData\vuegraf-master\src\vuegraf\vuegraf.py", line 12, in <module>
           import influxdb
     ModuleNotFoundError: No module named 'influxdb'

That portion of the code is:
# InfluxDB v1
import influxdb

# InfluxDB v2
import influxDb2

from pyemvue import PyEmVue
from pyemvue.enums import Scale, Unit

Any thoughts?  Is this supported to run on Windows?

Thanks!

Mark

**Comment by jertel** [comment, marked as answer]
Hello, it's a little tough to get a sense of time from your post. Are you saying you originally had the influxdb module error, fixed it, and now have the signal error? If so, good, I'm tracking with you. In that case, it looks like Vuegraf has some complications when running directly on Windows, since Linux signals are slightly different from Windows signals. One way to eliminate Windows related issues is to use Docker. The README has instructions on how to run Vuegraf on Docker. Is that an option for you?

**Comment by mjank72** [comment]
Sorry about that.  You have it correct.  I fixed the influxdb module error and then ran into the signal error.  I did some reading and saw that, as you point out, Windows doesn't handle the SIGHUP call.  As a test, I commented that line out and it executed successfully and added rows to my Influx db.  The downside is that, in order to stop the program, I had to close the window instead of using Ctrl-C.

I'll look into running a Docker instance, but I'm just testing for now.  If that's the only downside (closing the window to stop the program) I may stick with what I have, but we'll see.  Thanks for the response and for taking the time to put all this together!

Mark

**Reply by jertel** [reply]
Good debugging there. If that's the only problem line I can adjust that code to deal with it more gracefully. Let me know how it goes.

**Reply by mjank72** [reply]
Thanks for the help!  I switched to a WLS2 instance for InfluxDB2 and this program on my PC running Ubuntu and it's working great!  Quick question.  Does this download the minute data or second data?  If not the second data, is there a way (or a mod I can make to the code) in order to have it pull the second data?

**Reply by jertel** [reply]
Glad to hear you got it working! Read this discussion for more information on the per-minute and per-second data: 
#57

**Comment by mjank72** [comment]
That explains a lot for me!  Thanks!
